core/js/src/test/scala/com/lynbrookrobotics/potassium/JSClockTest.scala

```diff
@@ -42,4 +42,17 @@ class JSClockTest extends AsyncFunSuite {
 
     future.map(t => assert(t <= 250))
   }
+
+  test("Single execution can be canceled") {
+    val testDonePromise = Promise[Unit]()
+    val future = testDonePromise.future
+    var executed = false
+
+    JSClock.singleExecution(Milliseconds(500)) {
+      throw new IllegalStateException("This code should not have run")
```

Comment:
Because this is being thrown inside a separate event, it won't affect the results of this unit test. You should be setting the `executed` variable here.